yogstation/code/modules/antagonists/flockmind/abilities.dm

```diff
@@ -0,0 +1,251 @@
+		return
+	var/mob/living/simple_animal/hostile/flockdrone/FD = owner
+	FD.EjectPilot()
+
+/datum/action/cooldown/flock/enemi
+	name = "Designate Enemy"
+	desc = "Alert your Flock that someone is definitely an enemy of your flock. NPC drones will fire lethal lasers at them regardles of conditions."
+	action = /datum/flock_command/enemy_of_the_flock
+	button_icon_state = "designate_enemy"
+
+/datum/action/cooldown/flock/flocktrace
+	name = "Partition Mind"
+	desc = "Summon a flocktrace, that will be able to pilot flockdrones. Requires 100 compute."
+	cooldown_time = 60 SECONDS
+	button_icon_state = "awaken_drone"
+	var/waiting = FALSE
+
+/datum/action/cooldown/flock/flocktrace/Trigger()
+	if(waiting || !isflockmind(owner))
+		return
+	var/datum/team/flock/team = get_flock_team()
+	if(team.get_compute(TRUE) < 100)
+		owner.balloon_alert(owner, "Not enough compute")
+		return
+	waiting = TRUE
+	to_chat(owner, span_notice("You attempt to summon a Flocktrace..."))
+	var/list/candidates = pollGhostCandidates("Do you want to play as a flocktrace?", ROLE_FLOCKTRACE)
+	if(!candidates.len || team.get_compute(TRUE) < 100) //Check again for the amount of compute, in case if it changed while we were polling for ghosts
+		waiting = FALSE
+		owner.balloon_alert(owner, "Summon failed, try again later.")
+		return
+	
+	var/mob/dead/selected = pick(candidates)
+	var/datum/mind/player_mind = new /datum/mind(selected.key)
+	player_mind.active = TRUE
+	var/mob/camera/flocktrace/FT = new (get_turf(owner))
+	player_mind.transfer_to(FT)
+	player_mind.assigned_role = "Flocktrace"
+	player_mind.special_role = "Flocktrace"
+	team.add_member(player_mind)
+	owner.balloon_alert(owner, "Summoned sucessfully")
+	message_admins("[ADMIN_LOOKUPFLW(FT)] has been made into a Flocktrace by [ADMIN_LOOKUPFLW(owner)]'s [name] ability.")
+	log_game("[key_name(FT)] was spawned as a Flocktrace by [key_name(owner)]'s [name] ability.")
+	waiting = FALSE
+	StartCooldown()
+
+/datum/action/cooldown/flock/repair_burst
+	name = "Concentrated Repair Burst"
+	desc = "Fully heal a drone through acceleration of its repair processes."
+	action = /datum/flock_command/repair
+	cooldown_time = 20 SECONDS
+	button_icon_state = "heal_drone"
+
+/datum/action/cooldown/flock/door_open
+	name = "Gatecrash"
+	desc = "Open any non-bolted and AI controlable airlocks near you."
+	cooldown_time = 10 SECONDS
+	button_icon_state = "open_door"
+
+/datum/action/cooldown/flock/door_open/Trigger()
+	var/list/targets = list()
+	for(var/obj/machinery/door/airlock/A in range(10, get_turf(owner)))
+		if(A.canAIControl())
+			targets += A
+	if(length(targets))
+		playsound(owner, 'sound/misc/flockmind/flockmind_cast.ogg', 80, 1)
+		to_chat(owner, span_notice("You force open all the airlocks around you."))
+		StartCooldown()
+		sleep(1.5 SECONDS)
+		for(var/obj/machinery/door/airlock/A in targets)
+			A.open()
+	else
+		owner.balloon_alert(owner, "No valid targets")
+		return
```

Comment:
switch this around and remove the else (!targets.len)

Reply:
also i'm pretty sure balloon alerts (on tg atleast) use full on lowercase

Reply:
Done. Also no, on TG balloon alerts are sometimes are not fully on lowercase.